.gitignore

```diff
@@ -6,3 +6,4 @@ assets/
 *.so.*
 demo_project/
 Cargo.lock
+.vscode/
```

Comment:
Personally I agree with this, but iirc @Salzian had some reservations about this

Reply:
Maybe add both like in the Bevy repo:
```
/.idea
/.vscode
```

Reply:
I don't see a problem with it, but id love to see the conversation with @Salzian if its public.

Reply:
`.vscode` and `.idea` sometimes serve valuable metadata about the repository between developers. They usually have their own nested `.gitignore` to only share the important bits. So feel free to check in the `.vscode` folder.

Reply:
I'm gonna undo the changes, feels out of scope for the PR.